samples/sensor/magn_polling/sample.yaml

```diff
@@ -1,8 +1,26 @@
+common:
+  harness: sensor
+  tags: sensors
 sample:
-  name: BMC150 Magnetometer Sample
+  name: Magnetometer Polling Sample
 tests:
-  sample.sensor.magn_polling:
-    harness: sensor
-    tags: sensors
-    depends_on: i2c
+  sample.sensor.magn_polling.bmc150_magn:
     filter: dt_compat_enabled("bosch,bmc150_magn")
+    extra_args: MAGN_POLLING_COMPAT=bosch_bmc150_magn
+  sample.sensor.magn_polling.bmm150:
+    filter: dt_compat_enabled("bosch,bmm150")
+    extra_args: MAGN_POLLING_COMPAT=bosch_bmm150
+  sample.sensor.magn_polling.fxos8700:
+    filter: dt_compat_enabled("nxp,fxos8700")
+    extra_args: MAGN_POLLING_COMPAT=nxp_fxos8700
+    extra_configs:
+      - CONFIG_FXOS8700_MODE_HYBRID=y
+  sample.sensor.magn_polling.hmc5883l:
+    filter: dt_compat_enabled("honeywell,hmc5883l")
+    extra_args: MAGN_POLLING_COMPAT=honeywell_hmc5883l
+  sample.sensor.magn_polling.lis2mdl:
+    filter: dt_compat_enabled("st,lis2mdl")
+    extra_args: MAGN_POLLING_COMPAT=st_lis2mdl
+  sample.sensor.magn_polling.lsm303dlhc-magn:
+    filter: dt_compat_enabled("st,lsm303dlhc-magn")
+    extra_args: MAGN_POLLING_COMPAT=st_lsm303dlhc_magn
```

Comment:
note: wouldn't it be a better idea to rely on aliases like we do for LEDs, buttons, etc.? Something like

```
aliases {
    magn-0 = &mymagnetometer;
};
```

then in the sample do

```
... DEVICE_DT_GET(DT_ALIAS(magn_0));
```

Reply:
My initial thought was that I might want to try to support multiple driver instances and have the application read all of the available sensors, but that might overcomplicate the sample. I gave your alias idea a try and I like how it turned out. Please have look at the new version.